lcls_tools/common/image/roi.py

```diff
@@ -0,0 +1,109 @@
+from abc import ABC, abstractmethod
+from typing import (
+    List,
+)
+import numpy as np
+from matplotlib import patches
+from pydantic import BaseModel, PositiveFloat, Field
+
+
+class ROI(BaseModel, ABC):
+    roi_type: str
+    center: List[PositiveFloat]
+
+    @abstractmethod
+    def crop_image(self, img, **kwargs) -> np.ndarray:
+        """crop image using ROI"""
+        pass
+
+    @abstractmethod
+    def get_patch(self):
```

Comment:
Patches are plotting methods used to draw shapes on matplotlib graphs. I'm going to remove this, since we removed plotting in other parts of this PR.